hack/lib/build/version.sh

```diff
@@ -0,0 +1,121 @@
+			# version or whether the tree is between two tagged versions.
+			if [[ "${OS_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)(\.[0-9]+)*([-].*)?$ ]]; then
+				OS_GIT_MAJOR=${BASH_REMATCH[1]}
+				OS_GIT_MINOR=${BASH_REMATCH[2]}
+				OS_GIT_PATCH=${BASH_REMATCH[3]}
+				if [[ -n "${BASH_REMATCH[5]}" ]]; then
+					OS_GIT_MINOR+="+"
+				fi
+			fi
+
+			# This translates the "git describe" to an actual semver.org
+			# compatible semantic version that looks something like this:
+			#   v1.1.0-alpha.0.6+84c76d1-345
+			OS_GIT_VERSION=$(echo "${OS_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{7,40\}\)$/\+\2-\1/")
+			# If this is an exact tag, remove the last segment.
+			OS_GIT_VERSION=$(echo "${OS_GIT_VERSION}" | sed "s/-0$//")
+			if [[ "${OS_GIT_TREE_STATE}" == "dirty" ]]; then
+				# git describe --dirty only considers changes to existing files, but
+				# that is problematic since new untracked .go files affect the build,
+				# so use our idea of "dirty" from git status instead.
+				OS_GIT_VERSION+="-dirty"
+			fi
+		fi
+	fi
+
+	#os::build::version::etcd_vars
+	#os::build::version::kubernetes_vars
+}
+readonly -f os::build::version::git_vars
+
+function os::build::version::etcd_vars() {
+	ETCD_GIT_VERSION=$(go run "${OS_ROOT}/tools/godepversion/godepversion.go" "${OS_ROOT}/Godeps/Godeps.json" "github.com/coreos/etcd/etcdserver" "comment")
+	ETCD_GIT_COMMIT=$(go run "${OS_ROOT}/tools/godepversion/godepversion.go" "${OS_ROOT}/Godeps/Godeps.json" "github.com/coreos/etcd/etcdserver")
+}
+readonly -f os::build::version::etcd_vars
+
+# os::build::version::kubernetes_vars returns the version of Kubernetes we have
+# vendored.
+function os::build::version::kubernetes_vars() {
+	KUBE_GIT_VERSION=$(go run "${OS_ROOT}/tools/godepversion/godepversion.go" "${OS_ROOT}/Godeps/Godeps.json" "k8s.io/kubernetes/pkg/api" "comment")
+	KUBE_GIT_COMMIT=$(go run "${OS_ROOT}/tools/godepversion/godepversion.go" "${OS_ROOT}/Godeps/Godeps.json" "k8s.io/kubernetes/pkg/api")
+
+	# This translates the "git describe" to an actual semver.org
+	# compatible semantic version that looks something like this:
+	#   v1.1.0-alpha.0.6+84c76d1142ea4d
+	#
+	# TODO: We continue calling this "git version" because so many
+	# downstream consumers are expecting it there.
+	KUBE_GIT_VERSION=$(echo "${KUBE_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{7,40\}\)$/\+${OS_GIT_COMMIT:-\2}/")
+
+	# Try to match the "git describe" output to a regex to try to extract
+	# the "major" and "minor" versions and whether this is the exact tagged
+	# version or whether the tree is between two tagged versions.
+	if [[ "${KUBE_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)\. ]]; then
+		KUBE_GIT_MAJOR=${BASH_REMATCH[1]}
+		KUBE_GIT_MINOR="${BASH_REMATCH[2]}+"
+	fi
+}
+readonly -f os::build::version::kubernetes_vars
+
+# Saves the environment flags to $1
+function os::build::version::save_vars() {
+	cat <<EOF
+OS_GIT_COMMIT='${OS_GIT_COMMIT-}'
```

Comment:
these are not the same variables as ART sets - they have named it SOURCE_GIT_COMMIT now - I suppose this breaks versioning our binaries - maybe origin was special